front/app/components/admin/Graphs/_components/Container.tsx

```diff
@@ -62,14 +63,33 @@ const Container = ({
     resizeRef.current = node;
   };
 
-  const parsedHeight =
-    typeof height === 'number' && legend?.maintainGraphHeight
-      ? height + (legendDimensions?.height ?? 0)
-      : height;
+  const rightLegend = legend?.position?.includes('right');
+
+  const parsedWidth = !rightLegend
+    ? width
+    : typeof width === 'number' && legend?.maintainGraphSize
+    ? width + (legendDimensions?.width ?? 0) + (legend.marginLeft ?? 0)
+    : width;
+
+  const parsedHeight = rightLegend
+    ? height
+    : typeof height === 'number' && legend?.maintainGraphSize
+    ? height + (legendDimensions?.height ?? 0)
```

Comment:
This should also have `legend.marginTop`- bug in my code :) Also if the legend has no margin the fallback should be the default margin, not 0. Cleaned this up in the other PR